weave/trace_server/interface/builtin_object_classes/llm_structured_model.py

```diff
@@ -56,10 +86,205 @@ class LLMStructuredCompletionModelDefaultParams(BaseModel):
     # response_format_schema: dict | None = None
 
 
-class LLMStructuredCompletionModel(base_object_def.BaseObject):
+def cast_to_message_list(obj: Any) -> list[Message]:
+    if isinstance(obj, Message):
+        return [obj]
+    elif isinstance(obj, dict):
+        return [Message.model_validate(obj)]
+    elif isinstance(obj, str):
+        return [Message(content=obj, role="user")]
+    elif isinstance(obj, list):
+        return [cast_to_message(item) for item in obj]
+    raise TypeError("Unable to cast to Message")
+
+
+def cast_to_message(obj: Any) -> Message:
+    if isinstance(obj, Message):
+        return obj
+    elif isinstance(obj, dict):
+        return Message.model_validate(obj)
+    elif isinstance(obj, str):
+        return Message(content=obj, role="user")
+    raise TypeError("Unable to cast to Message")
+
+
+def cast_to_llm_structured_model_params(
+    obj: Any,
+) -> LLMStructuredCompletionModelDefaultParams:
+    if isinstance(obj, LLMStructuredCompletionModelDefaultParams):
+        return obj
+    elif isinstance(obj, dict):
+        return LLMStructuredCompletionModelDefaultParams.model_validate(obj)
+    raise TypeError("Unable to cast to LLMStructuredCompletionModelDefaultParams")
+
+
+MessageListLike = Annotated[list[Message], BeforeValidator(cast_to_message_list)]
+MessageLike = Annotated[Message, BeforeValidator(cast_to_message)]
+LLMStructuredModelParamsLike = Annotated[
+    LLMStructuredCompletionModelDefaultParams,
+    BeforeValidator(cast_to_llm_structured_model_params),
+]
+
+
+class LLMStructuredCompletionModel(Model):
     # <provider>/<model> or ref to a provider model
     llm_model_id: Union[str, base_object_def.RefStr]
 
     default_params: LLMStructuredCompletionModelDefaultParams = Field(
         default_factory=LLMStructuredCompletionModelDefaultParams
     )
+
+    @op
+    def predict(
+        self,
+        user_input: MessageListLike = [],
+        config: Optional[LLMStructuredModelParamsLike] = None,
+        **template_vars: Any,
+    ) -> Union[Message, str, dict[str, Any]]:
+        """
+        Generates a prediction by preparing messages (template + user_input)
+        and calling the LLM completions endpoint with overridden config, using the provided client.
+
+        Args:
+            user_input: The user input messages
+            config: Optional configuration to override default parameters
+            **template_vars: Variables to substitute in the messages template using {variable_name} syntax
+        """
+        current_client = require_weave_client()
```

Comment:
I might use `get_weave_client` and then more gracefully raise an informative error in the case that one is not initialized.